src/features/device-provisioning/gracefully-deny-conflicting-registrations.ts

```diff
@@ -0,0 +1,101 @@
+import _ from 'lodash';
+import { sbvrUtils, errors, permissions } from '@balena/pinejs';
+import {
+	API_KEY_EXISTS_CACHE_TIMEOUT,
+	DEVICE_EXISTS_CACHE_TIMEOUT,
+} from '../../lib/config';
+import { createMultiLevelStore } from '../../infra/cache';
+
+const { ConflictError } = errors;
+const { api } = sbvrUtils;
+
+const checkDeviceExistsQuery = _.once(() =>
```

Comment:
Don't we already have a mechanism for checking devices exist?

Reply:
We do but the other one the exists/isFrozen one which is unnecessarily expensive if it has to run.
Moreover for this case we do not care to cache the result if the device does not exist.